### DISCUSSION TITLE
Error encountered in NeighborLoader for own dataset

### DISCUSSION THREAD

**Discussion by xubingze** [pyg-team/pytorch_geometric#9017] [Q&A] [answered]
Hi there! I am trying to do node classification for my own dataset. There are 7000+ nodes in the network, but only 1000 of them contains label. So I define mask for the nodes that have labels.

```
train_loader = NeighborLoader(data_stock, input_nodes=torch.tensor(data_stock.train_mask, dtype=torch.bool),
                              num_neighbors=[3,2], batch_size=256,
                              directed=False, shuffle=True)
print(data_stock)
print(data_stock.train_mask)
print(torch.tensor(data_stock.train_mask, dtype=torch.bool))
for batch in train_loader:
    print(batch)
```
However, I encounter this error and I have no ideas why this happen since I have converted the type of data_stock.train_mask. Thank you very much for your help!
<img width="984" alt="Screen Shot 2024-03-04 at 10 12 12 PM" src="https://github.com/pyg-team/pytorch_geometric/assets/26111084/6f70330d-077b-4ac4-8269-bcae612dd866">

**Comment by wsad1** [comment]
Convert all your masks in `data_stock` like `train_mask`to torch tensors. That should fix this.

**Reply by xubingze** [reply]
Thank you very much for your reply! I tried your suggestion, but the error still occurred. I plan to use NeighborLoader only in training not in testing process, so I am not sure why the val_mask or test_mask can lead to this error message. Here is the screenshot: 
<img width="1361" alt="Screen Shot 2024-03-07 at 11 14 26 PM" src="https://github.com/pyg-team/pytorch_geometric/assets/26111084/f3ae1c46-1ddb-4fd5-bcc0-d705e5c48afa">

**Reply by wsad1** [reply, marked as answer]
Are there any other `list`s in `data_stock`. Whats the type of `x`, `edge_index` and `y`?

**Reply by xubingze** [reply]
Yes, I checked the type of x, edge_index and y. The first two are tensor type, but the y are generated from a list (each node is classified as a number between 0 and 5). After I converted y as the type of torch.int64, the error is resolved. Thank you very much!
